### ISSUE TITLE
Question: Do Rules Transfer State or Action

### ISSUE THREAD

**Issue by rei-vilo** [mozilla-iot/gateway#693]
Very similar to the question about [Discussion: Possible Confusion between State and Action on Devices](https://discourse.mozilla.org/t/discussion-possible-confusion-between-state-and-action-on-devices/25680/12) but this time about **Rules**.

I conducted the following tests with one push button (GPIO 18) and a LED (CPIO 17) and a rule, for example, `Button ON => LED OFF` or `ON>OFF`. 

<img width="400" alt="capture 2018-02-19 a 21 04 46" src="https://user-images.githubusercontent.com/951776/36395963-16d0396a-15bc-11e8-91e4-1e9ef6f8b9a5.png">

The protocol for each test includes:
+ Define one single rule among `ON>ON`, `ON>OFF`, `OFF>ON`, `OFF>OFF`
+ Conduct 2 sequences, 
    + First sequence with initial state of the LED on, and 
    + Second sequence with initial state of the LED off.
+ Report 4 consecutive states
    + Initial state of the LED,
    + State of the LED when button pressed,
    + State of the LED when button released,
    + Final state of the LED.

I don't understand why, with some rules, the final state differs from the initial (cases `OFF>OFF` Sequence 2 and `OFF>ON` Sequence 1) while for the other rules, final state resumes with initial state.

![mozilla rules](https://user-images.githubusercontent.com/951776/36396325-c61b7ae6-15bd-11e8-970f-f4e837667af8.jpg)

In the case of the `ON>ON` rule, I've only asked for `If Button 18 is on then turn LED 17` but not for `If Button 18 is off then turn LED 17 off`. 

Changing `"17": { "value": 1 }` for `"17": { "value": 0 }` gets the same results.

Could you please explain the rationale behind the rules?

----

For reference: 

```
$ cd ~/mozilla-iot/gateway/tools
$ ./config-editor.py gpio-adapter
{
  "pins": {
    "17": {
      "direction": "out",
      "name": "LED 17",
      "value": 1
    },
    "18": {
      "debounce": 10,
      "direction": "in",
      "edge": "both",
      "name": "Button 18"
    }
  }
}
```

**Comment by hobinjk** [comment]
The way the logic of rules works is a bit less than intuitive right now since there is no user interface method to differentiate between what I would call a "pulse" and a "set" effect.

Right now the intended design is that the rule saves the original state of the effected device and reverts to it when the rule is done executing (unless it's a time trigger). Based on your data it looks like the off trigger isn't behaving as intended for some reason. I'll try to track down what's going on. Thanks for the report!

**Comment by hobinjk** [comment]
This behavior should be fixed by #702
